srml/treasury/src/lib.rs

```diff
@@ -346,29 +348,12 @@ impl<T: Trait> Module<T> {
 
 impl<T: Trait> OnUnbalanced<NegativeImbalanceOf<T>> for Module<T> {
 	fn on_unbalanced(amount: NegativeImbalanceOf<T>) {
+		let numeric_amount = amount.peek();
+
 		// Must resolve into existing but better to be safe.
 		let _ = T::Currency::resolve_creating(&Self::account_id(), amount);
-	}
-}
 
-/// Mint extra funds for the treasury to keep the ratio of portion to total_issuance equal
-/// pre dilution and post-dilution.
-///
-/// i.e.
-/// ```nocompile
-/// portion / total_issuance_before_dilution ==
-///   (portion + minted) / (total_issuance_before_dilution + minted_to_treasury + minted)
-/// ```
-impl<T: Trait> OnDilution<BalanceOf<T>> for Module<T> {
```

Comment:
If we are to not have this trait and just use imbalances for it (as it seems here), I think it can be removed as a trait (have not done a clone build but should already give some warnings afaiu)